### PR TITLE
Drawing regions from list of points

### PR DESCRIPTION
**Pull request by jgsdavies**

### CODE REVIEW

File: src/ansys/motorcad/core/geometry.py [hunk 1]
```diff
@@ -39,6 +40,14 @@ def __eq__(self, other):
             and self.entities == other.entities
         )
 
+    @classmethod
+    def from_coordinate_list(cls):
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/geometry.py:44]: Need to refactor geometry.py before this is possible. Think this needs doing anyway since this is getting huge

File: src/ansys/motorcad/core/geometry.py [hunk 2]
```diff
@@ -526,8 +575,8 @@ class Entity(object):
 
     def __init__(self, start, end):
         """Initialise Entity."""
-        self.start = start
-        self.end = end
+        self.start = deepcopy(start)
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/geometry.py:578]: Trying to avoid people accidentally editing entities if they mutate a coordinate

@matthewAnsys thoughts?

File: src/ansys/motorcad/core/geometry.py [hunk 3]
```diff
@@ -728,6 +799,66 @@ def coordinate_on_entity(self, coordinate):
...
+            Angle of rotation in degrees.
+
+        Returns
+        -------
+        Line
+        """
+        return Line(self.start.rotate(centre_point, angle), self.end.rotate(centre_point, angle))
+
+    def _line_equation(self):
+        """Get value of coefficients of line in format A*x + B*y = C."""
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/geometry.py:819]: Useful because  A inf/undefined for vertical lines when using (y = Ax + B)

File: src/ansys/motorcad/core/geometry.py [hunk 4]
```diff
@@ -1162,3 +1351,37 @@ def rt_to_xy(radius, theta):
     y = coordinates_complex.imag
 
     return x, y
+
+
+class _Orientation(Enum):
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/geometry.py:1356]: Kept these as internal functions for now

File: src/ansys/motorcad/core/geometry.py [hunk 5]
```diff
@@ -509,7 +518,47 @@ def mirror(self, mirror_line):
...
+        """
+        angle_r = radians(angle)
+        x_new = centre_point.x + (
+            (cos(angle_r) * (self.x - centre_point.x)) + (-sin(angle_r) * (self.y - centre_point.y))
+        )
+        y_new = centre_point.y + (
+            (sin(angle_r) * (self.x - centre_point.x)) + (cos(angle_r) * (self.y - centre_point.y))
+        )
+
+        return Coordinate(x_new, y_new)
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/geometry.py:545]: @matthewAnsys 

Decided to return new coordinate instead of mutating existing.

Not sure I like this but I think mutating coordinate is likely to confuse beginner Python users

File: src/ansys/motorcad/core/geometry_drawing.py
Comment (jgsdavies) [inline src/ansys/motorcad/core/geometry_drawing.py:1]: This has been done in https://github.com/ansys/pymotorcad/pull/246

and should disappear from this PR when this other PR has been merged. No need to review this file here

File: src/ansys/motorcad/core/geometry_fitting.py [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+from ansys.motorcad.core.geometry import Arc, Coordinate, EntityList, Line
+
+
+class _EntityType(Enum):
+    line = 0
+    arc = 1
+
+
+class _TestEntity:
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/geometry_fitting.py:14]: Used to check if list of points is within an acceptable tolerance for an entity

File: tests/test_geometry_drawing.py
Comment (jgsdavies) [inline tests/test_geometry_drawing.py:1]: Will be reviewed as part of https://github.com/ansys/pymotorcad/pull/246

File: tests/test_geometry_fitting.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
+from ansys.motorcad.core.geometry import Arc, Coordinate, Line
+from ansys.motorcad.core.geometry_fitting import return_entity_list
+
+
+def test_return_entity_list():
```
Comment (jgsdavies) [inline tests/test_geometry_fitting.py:5]: basic test - will almost certainly need to add to this to catch edge cases

File: src/ansys/motorcad/core/geometry_fitting.py [hunk 8]
```diff
@@ -0,0 +1,200 @@
+"""Unit containing functions for drawing geometry from a list of points."""
```
Comment (matthewAnsys) [inline src/ansys/motorcad/core/geometry_fitting.py:1]:
```suggestion
"""Unit containing functions for creating geometry using from a list of points."""
```

File: src/ansys/motorcad/core/geometry_fitting.py [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+
+from copy import copy
+from enum import Enum
+
+from ansys.motorcad.core.geometry import Arc, Coordinate, EntityList, Line
+
+
+class _EntityType(Enum):
```
Comment (matthewAnsys) [inline src/ansys/motorcad/core/geometry_fitting.py:9]: missing docstring

File: src/ansys/motorcad/core/geometry_fitting.py (hunk 6 above)
Comment (matthewAnsys) [reply src/ansys/motorcad/core/geometry_fitting.py:14]: can you add a docstring

File: src/ansys/motorcad/core/geometry_fitting.py [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+    line = 0
+    arc = 1
+
+
+class _TestEntity:
```
Comment (matthewAnsys) [inline src/ansys/motorcad/core/geometry_fitting.py:14]: Not sure I partically like this class name, but unsure of what it could be updated to

**Review by matthewAnsys — CHANGES_REQUESTED** [review]

File: src/ansys/motorcad/core/geometry.py (hunk 5 above)
Comment (jgsdavies) [reply src/ansys/motorcad/core/geometry.py:567]: I personally agree but trying to avoid confusion with beginners.

```python

# mutating correct way
p1 = Coordinate()
p2 = copy(p1)
p2.rotate(90)

# suspect most people will try
p1 = Coordinate()
p2 = p1.rotate(90)
# p1 is p2 (confusing?)

# vs

p1 = Coordinate()
p2 = p1.rotate(90)

# p1 and p2 are different objects - works as expected
```

File: src/ansys/motorcad/core/geometry_fitting.py (hunk 10 above)
Comment (jgsdavies) [reply src/ansys/motorcad/core/geometry_fitting.py:14]: Me neither - open to suggestions :)

File: src/ansys/motorcad/core/geometry_fitting.py (hunk 9 above)
Comment (jgsdavies) [reply src/ansys/motorcad/core/geometry_fitting.py:9]: See other comment on private methods - added a short one anyway

File: src/ansys/motorcad/core/geometry.py [hunk 11]
```diff
@@ -509,7 +540,47 @@ def mirror(self, mirror_line):
...
+
+    @classmethod
+    def from_polar_coords(cls, radius, theta):
```
[the remark this answers is not in the document]
Comment (jgsdavies) [reply src/ansys/motorcad/core/geometry.py:570]: This is returning a Coordinate() object from polar coords. Basically an overloaded init. I think this name makes sense

File: src/ansys/motorcad/core/geometry.py [hunk 12]
```diff
@@ -509,7 +540,47 @@ def mirror(self, mirror_line):
...
+        """Create Motor-CAD coordinate from polar coordinates.
```
[the remark this answers is not in the document]
Comment (jgsdavies) [reply src/ansys/motorcad/core/geometry.py:571]: made this a bit clearer

**Review by matthewAnsys — APPROVED** [review]
